### PR TITLE
own: repo and directory level drilldown pages for owners and signals

### PR DESCRIPTION
**Pull request by coury-clark**
Closes [50799](https://github.com/sourcegraph/sourcegraph/issues/50799)
Adds repo and directory level drilldown for owners and ownership signals.

## Test plan

1. Navigate to a repository and click on "ownership" -> should navigate to the ownership page for the repo root

CleanShot 2023-06-01 at 16 19 35@2x

2. Click on "show more" from the ownership box from a directory view -> should navigate to the ownership page for that directory path
CleanShot 2023-06-01 at 16 18 59@2x

4. Click on edit ownership -> should navigate to the edit ownership page for the same repository
CleanShot 2023-06-01 at 16 20 58@2x

### CODE REVIEW

File: client/web/src/repo/blob/own/OwnerList.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
+import * as React from 'react'
```
Comment (coury-clark) [inline client/web/src/repo/blob/own/OwnerList.tsx:1]: Not fully sure why this file is showing a diff after merging main

File: sg.config.yaml [hunk 2]
```diff
@@ -1211,7 +1211,6 @@ commandsets:
       - gitserver-0
       - gitserver-1
       - searcher
-      - symbols
```
Comment (cbart) [inline sg.config.yaml:1214]: Was this intentional in this PR?

**Review by cbart — APPROVED** [review]
:shipit: 

Regarding the UI, I'd just add the breadcrumbs for the tree view if we're jumping from a directory to ownership page.

Screenshot 2023-06-02 at 9 33 44 AM

**Comment by cbart** [comment]
Similar here in the Upload CODEOWNERS file

Screenshot 2023-06-02 at 9 58 07 AM

**Review by sashaostrikov — APPROVED** [review]
:shipit:     :shipit:  :shipit:  :shipit::shipit::shipit::shipit:  :shipit::shipit::shipit::shipit:
:shipit::shipit:  :shipit:  :shipit:  :shipit:       :shipit:   
:shipit: :shipit: :shipit:  :shipit:  :shipit:       :shipit::shipit::shipit: 
:shipit:  :shipit::shipit:  :shipit:  :shipit:       :shipit:    
:shipit:     :shipit:  :shipit:  :shipit::shipit::shipit::shipit:  :shipit::shipit::shipit::shipit:

File: client/web/src/repo/blob/own/OwnerList.tsx (hunk 1 above)
Comment (coury-clark) [reply client/web/src/repo/blob/own/OwnerList.tsx:1]: Yeah, these changes are from a previous PR

File: client/web/src/enterprise/repo/enterpriseRepoContainerRoutes.tsx [hunk 3]
```diff
@@ -43,8 +44,13 @@ export const enterpriseRepoContainerRoutes: readonly RepoContainerRoute[] = [
         render: context => <RepositoryBatchChangesArea {...context} />,
     },
     {
-        path: '/-/own',
+        path: '/-/own/*',
         condition: ({ isSourcegraphDotCom }) => !isSourcegraphDotCom,
         render: context => <RepositoryOwnPage {...context} />,
     },
+    {
+        path: '/-/own/edit',
```
[the remark this answers is not in the document]
Comment (coury-clark) [reply client/web/src/enterprise/repo/enterpriseRepoContainerRoutes.tsx:52]: I didn't encounter any issues, but I'm also not an expert! I'll see if anyone on the frontend team can confirm for me

File: client/web/src/repo/blob/own/TreeOwnershipPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import { OwnerList } from './OwnerList'
+
+import styles from './FileOwnershipPanel.module.scss'
+
+export const TreeOwnershipPanel: React.FunctionComponent<
+    {
+        repoID: string
+        revision?: string
+        filePath: string
+    } & TelemetryProps
```
[the remark this answers is not in the document]
Comment (coury-clark) [reply client/web/src/repo/blob/own/TreeOwnershipPanel.tsx:23]: sure, done!

**Comment by coury-clark** [comment]
@cbart I've updated the single level breadcrumbs for now. Kelli gave me a way to do nested breadcrumbs, but it's a little janky, so I'll follow up in a new PR and see if we really want that

File: client/web/src/enterprise/repo/enterpriseRepoContainerRoutes.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (coury-clark) [reply client/web/src/enterprise/repo/enterpriseRepoContainerRoutes.tsx:52]: after following up with Vova, this should be fine

**Pull request merged**